packages/location-management/src/components/LocationUnitAddEdit/Form.tsx

```diff
@@ -0,0 +1,274 @@
+import * as Yup from 'yup';
+import React from 'react';
+import { SubmitButton, Form as AntForm, Input, Radio, Select, TreeSelect } from 'formik-antd';
+import { notification, Button } from 'antd';
+import { history } from '@onaio/connected-reducer-registry';
+import { getUser, User } from '@onaio/session-reducer';
+import { OpenSRPService } from '@opensrp/server-service';
+import { getAccessToken } from '@onaio/session-reducer';
+import { Formik } from 'formik';
+import {
+  LocationUnitPayloadPOST,
+  LocationUnitPayloadPUT,
+  LocationUnitStatus,
+  LocationUnitSyncStatus,
+  LocationUnitTag,
+} from '../../ducks/location-units';
+import { useSelector } from 'react-redux';
+import { Geometry } from 'geojson';
+import { API_BASE_URL, LOCATION_HIERARCHY, LOCATION_UNIT_POST_PUT } from '../../constants';
+import { v4 } from 'uuid';
+import { LocationTag } from '../../ducks/location-tags';
+import { ParsedHierarchyNode, RawOpenSRPHierarchy } from '../../ducks/types';
+
+export interface FormField {
+  name: string;
+  status: LocationUnitStatus;
+  type: string;
+  parentId?: string;
+  externalId?: string;
+  locationTags?: number[];
+  geometry?: string;
+}
+
+const defaultFormField: FormField = {
+  parentId: '',
+  name: '',
+  status: LocationUnitStatus.ACTIVE,
+  type: '',
+};
+
+export interface Props {
+  id?: string;
+  initialValue?: FormField;
+  locationtag: LocationTag[];
+  treedata: ParsedHierarchyNode[];
+}
+
+/** yup validations for practitioner data object from form */
+const userSchema = Yup.object().shape({
+  parentId: Yup.string().typeError('Parentid must be a String'),
+  name: Yup.string().typeError('Name must be a String').required('Name is Required'),
+  status: Yup.string().required('Status is Required'),
+  type: Yup.string().typeError('Type must be a String').required('Type is Required'),
+  externalId: Yup.string().typeError('External id must be a String'),
+  locationTags: Yup.array().typeError('location Tags must be an Array'),
+  geometry: Yup.string().typeError('location Tags must be a An String'),
+});
+const layout = { labelCol: { span: 8 }, wrapperCol: { span: 11 } };
+const offsetLayout = { wrapperCol: { offset: 8, span: 11 } };
+const status = [
+  { label: 'Active', value: LocationUnitStatus.ACTIVE },
+  { label: 'Inactive', value: LocationUnitStatus.INACTIVE },
+];
+
+/** removes empty undefined and null objects before they payload is sent to server
+ *
+ * @param {any} obj object to remove empty keys from
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function removeEmptykeys(obj: any) {
```

Comment:
why `any` here?

Reply:
the payload could be anything like maybe array of objects or just any object thats why i put any

Reply:
its hard to infer types for this 🤔💭

Reply:
@ashfahan Why can't we declare the type interface base on its usage below?

```
is_jurisdiction: true,
    properties: {
      geographicLevel: geographicLevel && geographicLevel >= 0 ? geographicLevel + 1 : 0,
      username: username,
      externalId: values.externalId,
      parentId: values.parentId ? values.parentId : '',
      name: values.name,
      // eslint-disable-next-line @typescript-eslint/camelcase
      name_en: values.name,
      status: values.status,
    },
    id: id ? id : v4(),
    syncStatus: LocationUnitSyncStatus.SYNCED,
    type: values.type,
    locationTags: locationTag,
    geometry: values.geometry ? (JSON.parse(values.geometry) as Geometry) : undefined,
```

Reply:
the problem is i am using recursion so that data type changes with the element passed in recursion

Reply:
@p-netm @ashfahan can we create this as a separate issue?  Please ping me when that is done
cc @dubdabasoduba